src/QsCompiler/CompilerOptimizations/Utils.fs

```diff
@@ -134,7 +167,7 @@ let rec internal takeWhilePlus1 (f: 'A -> bool) (l : list<'A>) =
 /// Returns all the "bottom-level" statements in the current scope.
 /// Bottom-level statements are defined as those that don't contain their own scopes.
 /// This function recurses through all the subscopes of the current scope.
-let rec private findAllBaseStatements (scope: QsScope): seq<QsStatementKind> =
+let rec internal findAllBaseStatements (scope: QsScope): seq<QsStatementKind> =
```

Comment:
As mentioned on a previous review, this needs to be replaced by a suitable transformation.